zetasql-toolkit-bigquery/src/main/java/com/google/zetasql/toolkit/catalog/bigquery/BigQueryCatalog.java

```diff
@@ -340,6 +357,27 @@ public void register(SimpleTable table, CreateMode createMode, CreateScope creat
     }
   }
 
+  private void registerWildcardTable(
```

Comment:
If we remove `buildCatalogNamesForWildcardTables()` as mentioned in the previous comment, we wouldn't need `registerWildcardTable()` either. We could use the existing `register()` method for `SimpleTable`s.